packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PreferencesStep/PreferencesStep.tsx

```diff
@@ -1,24 +1,76 @@
+import app from 'client/scripts/state';
+import {
+  useFetchSelfProfileQuery,
+  useUpdateProfileByAddressMutation,
+} from 'client/scripts/state/api/profiles';
 import {
   PreferenceTags,
   usePreferenceTags,
 } from 'client/scripts/views/components/PreferenceTags';
 import { CWText } from 'client/scripts/views/components/component_kit/cw_text';
 import { CWButton } from 'client/scripts/views/components/component_kit/new_designs/CWButton';
-import React from 'react';
+import AddressInfo from 'models/AddressInfo';
+import React, { useEffect, useRef } from 'react';
 import './PreferencesStep.scss';
 
 type PreferencesStepProps = {
   onComplete: () => void;
 };
 
 const PreferencesStep = ({ onComplete }: PreferencesStepProps) => {
-  const { selectedTags, toggleTagFromSelection } = usePreferenceTags({});
+  const initialTagsSet = useRef(false);
+  const { preferenceTags, setPreferenceTags, toggleTagFromSelection } =
+    usePreferenceTags({});
+
+  const { mutateAsync: updateProfile, isLoading: isUpdatingProfile } =
+    useUpdateProfileByAddressMutation();
+
+  const { data: profile, isLoading } = useFetchSelfProfileQuery({
+    apiCallEnabled: true,
+  });
+
+  useEffect(() => {
+    if (!isLoading && profile && !initialTagsSet.current) {
+      const profileTags = profile.tags;
```

Comment:
PreferencesStep is on user onboarding flow. Is there a case that there are gonna be selected tags already when this modal will pop up? Shouldn't this always start with empty array of setPreferenceTags?

Reply:
Actually there wont be a case like that, i added it because during testing, i already has selected tags. Lmk if you'd like me to remove

Reply:
yeah, let's not pollute the code, please remove